hedera-node/src/main/java/com/hedera/services/txns/token/TokenCreateTransitionLogic.java

```diff
@@ -131,6 +132,49 @@ private void transitionFor(TokenCreateTransactionBody op) {
 		txnCtx.setStatus(SUCCESS);
 	}
 
+	private ResponseCodeEnum autoEnableFeeCollectors(
+			TokenID created,
+			AccountID alreadyEnabledTreasury,
+			TokenCreateTransactionBody op
+	) {
+		/* TokenStore.associate() returns TOKEN_ALREADY_ASSOCIATED_TO_ACCOUNT if the
+		account-token relationship already exists. */
+		final Set<AccountID> customCollectorsEnabled = new HashSet<>();
+		customCollectorsEnabled.add(alreadyEnabledTreasury);
+		ResponseCodeEnum status = OK;
+		for (var fee : op.getCustomFees().getCustomFeesList()) {
+			if (fee.hasFractionalFee()) {
+				final var collector = fee.getFeeCollectorAccountId();
+				if (!customCollectorsEnabled.contains(collector)) {
+					status = autoEnableAccountForNewToken(collector, created, op);
+					if (status != OK) {
+						return status;
+					}
+					customCollectorsEnabled.add(collector);
+				}
+			}
+		}
+		return status;
+	}
+
+	private ResponseCodeEnum autoEnableAccountForNewToken(
+			AccountID id,
+			TokenID created,
+			TokenCreateTransactionBody op
+	) {
+		var status = store.associate(id, List.of(created));
+		if (status != OK) {
+			return status;
+		}
+		if (op.hasFreezeKey()) {
+			status = ledger.unfreeze(id, created);
+		}
+		if (status == OK && op.hasKycKey()) {
+			status = ledger.grantKyc(id, created);
+		}
```

Comment:
This part should be validated only once. Should extract it out, or the line 149 will do these validations again for all fee collectors.

Reply:
We want to do this for each fractional fee collector...So the point of L149 _is_ to repeat this for each `collector`. 

Am I missing something?

Reply:
Nvm, I misunderstood a few details. All is good!